settings.gradle.kts

```diff
@@ -0,0 +1,51 @@
+pluginManagement {
+    repositories {
+        mavenLocal()
+        mavenCentral()
+        gradlePluginPortal()
+    }
+
+    plugins {
+        val smithyGradleVersion : String by settings
+        id("software.amazon.smithy.gradle.smithy-jar") version smithyGradleVersion
+    }
```

Comment:
Previously we would pin the version in specific subprojects' `build.gradle.kts` like this:
```
plugins {
     id("software.amazon.smithy.gradle.smithy-jar") version smithyGradleVersion
}
		
```

However I found that the correct way to do version pinning for dependencies from `gradle.properties` is this.

This is also temporary while we move to the `versions.toml` approach like other projects and `buildSrc` based plugins in future PRs.

References:
 - https://docs.gradle.org/current/userguide/plugins.html#sec:plugin_management 
- Github issue on gradle side: https://github.com/gradle/gradle/issues/1697